src/Checkbox.js

```diff
@@ -1,175 +1,192 @@
-                        }
-
-                        .disabled {
-                            cursor: not-allowed;
-                            color: ${theme.disabled};
-                        }
-
-                        .focus {
-                            outline: 1px;
-                        }
-                    `}</style>
-                </label>
-            </Fragment>
+            <label
+                className={cx(className, {
+                    disabled,
+                    dense,
+                })}
+            >
+                <input
+                    type="checkbox"
+                    ref={this.ref}
+                    name={name}
+                    value={value}
+                    checked={checked}
+                    disabled={disabled}
+                    tabIndex={tabIndex}
+                    onChange={onChange}
+                    onFocus={onFocus}
+                    onBlur={onBlur}
+                />
+
+                <div className={cx('icon', { dense })}>
+                    {dense ? (
+                        <Dense className={classes} />
+                    ) : (
+                        <Regular className={classes} />
+                    )}
+                </div>
+
+                {label}
+
+                <style jsx>{`
+                    label {
+                        position: relative;
+                        display: flex;
+                        flex-direction: row;
+                        align-items: center;
+                        justify-content: flex-start;
+                        cursor: pointer;
+                        pointer-events: all;
+                        user-select: none;
+                        color: ${colors.grey900};
+                        padding-top: 4px;
+                        font-size: 16px;
+                        line-height: 20px;
+                    }
+
+                    label.dense {
+                        font-size: 14px;
+                        line-height: 16px;
+                        padding-top: 2px;
+                    }
+
+                    label.dense:first-of-type {
+                        padding-top: 4px;
+                    }
+
+                    label.disabled {
+                        cursor: not-allowed;
+                        color: ${theme.disabled};
+                    }
+                    input {
+                        position: absolute;
+                        opacity: 0;
+                        pointer-events: none;
+                    }
+
+                    .icon {
+                        pointer-events: none;
+                        user-select: none;
+                        margin-right: 5px;
+                        border: 2px solid transparent;
+                        padding: 1px;
+                        border-radius: 5px;
+                    }
+
+                    label.dense .icon {
+                        margin-right: 3px;
```

Comment:
Adding `border-radius: 4px` to `label.dense .icon` renders the correct focus border for dense.